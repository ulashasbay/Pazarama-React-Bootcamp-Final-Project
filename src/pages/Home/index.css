.hero {
  min-height: 100vh;
  background-image: linear-gradient(
    90deg,
    #ffffff 0%,
    #ffffff 55%,
    #fff0e8 55%,
    #fff0e8 100%
  );
}

.home-row {
  display: flex;
  flex-wrap: wrap;
  width: 80%;
  padding-top: 4rem;
  margin: -40px auto;
  justify-content: center;
}
.home-row h1 {
  font-size: 42px;
  margin-bottom: 40px;
  color: #0137f3;
  max-width: 520px;
}

.home-row p {
  font-size: 16px;
  margin-bottom: 30px;
  max-width: 520px;
}

.content-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 50%;
}

.btn-home {
  padding: 18px 34px;
  font-size: 18px;
  font-weight: 700;
  display: inline-block;
  max-width: 450px;
  margin-bottom: 24px;
  color: #fff;
  background-color: #0137f3;
  border: none;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.btn-home:hover {
  background: #03249e;
}

/* a {
  color: #007bff;
  text-decoration: none;
  background-color: transparent;
} */

.home-img {
  width: 100%;
  margin: 0 auto;
}

@media screen and (max-width: 820px) {
  .hero {
    background-image: linear-gradient(
      180deg,
      #ffffff 0%,
      #ffffff 55%,
      #fff0e8 55%,
      #fff0e8 100%
    );
  }
  .home-header {
    margin-bottom: 0;
  }

  .content-wrapper {
    max-width: 100% !important;
    margin-top: auto;
  }

  .btn-home {
    max-width: 600px;
  }

  .home-img {
    width: 100%;
    margin: 0 auto;
    padding: 2rem 0 2rem 0;
  }
}

@media screen and (min-width: 100px) {
  .home-row {
    padding-top: 6rem;
    margin: -40px auto;
  }
}
