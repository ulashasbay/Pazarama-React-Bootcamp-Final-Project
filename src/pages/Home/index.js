import React from "react";
import { useNavigate } from "react-router-dom";
import HomePageImg from "../../assets/images/homepage-img.png";
import "./index.css";

function Home() {
  let navigate = useNavigate();

  const handleClick = () => {
    navigate("/basvuru-olustur");
  };

  return (
    <div className="home-row">
      <div className="content-wrapper ">
        <img className="home-img" src={HomePageImg} alt="HomeImg" />
      </div>
      <div className="content-wrapper">
        <h1 className="home-header">Bizimle Çalışmak İster misiniz?</h1>
        <p>
          Pazarama&#39;da satış yapmak, yazılımcı ya da stajyer olarak işe
          başlamak için hemen başvur!
        </p>
        <button className="btn-home" onClick={handleClick}>
          Hemen Başvur!
        </button>
      </div>
    </div>
  );
}

export default Home;
