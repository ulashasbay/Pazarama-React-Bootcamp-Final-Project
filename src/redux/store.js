import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
