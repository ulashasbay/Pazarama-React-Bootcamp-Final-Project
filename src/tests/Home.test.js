import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "../pages/Home";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("HomePage Tests", () => {
  beforeEach(() => {
    render(<Home />, { wrapper: AllTheProviders });
  });

  test("Render image", () => {
    const homeImg = screen.getByAltText("HomeImg");
    expect(homeImg).toBeInTheDocument();
  });

  test("Render button", () => {
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
  });
});

test("Home renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
