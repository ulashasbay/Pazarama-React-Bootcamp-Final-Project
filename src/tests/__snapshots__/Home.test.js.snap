// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home-row"
>
  <div
    className="content-wrapper "
  >
    <img
      alt="HomeImg"
      className="home-img"
      src="homepage-img.png"
    />
  </div>
  <div
    className="content-wrapper"
  >
    <h1
      className="home-header"
    >
      Bizimle Çalışmak İster misiniz?
    </h1>
    <p>
      Pazarama'da satış yapmak, yazılımcı ya da stajyer olarak işe başlamak için hemen başvur!
    </p>
    <button
      className="btn-home"
      onClick={[Function]}
    >
      Hemen Başvur!
    </button>
  </div>
</div>
`;
